### PR TITLE
github: simplify main workflow (develop)

### PR DESCRIPTION
**Pull request by abn**
# Pull Request Check List

- [ ] ~~Added **tests** for changed code.~~
- [ ] ~~Updated **documentation** for changed code.~~

### CODE REVIEW

**Comment by abn** [comment]
@sdispater not sure how you want to handle this, we will have to have separate branch protection checks for master and develop until they are in sync.

**Comment by sdispater** [comment]
@abn Is it possible to make another PR in addition to this one that's targeting `master`?

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -8,120 +8,62 @@ jobs:
...
           pre-commit run --all-files
 
-  Linux:
+  Tests:
     needs: Linting
-    runs-on: ubuntu-latest
+    runs-on: ${{ matrix.os }}
     strategy:
       matrix:
+        os: [ubuntu-latest, macos-latest, windows-latest]
```
Comment (pradyunsg) [inline .github/workflows/main.yml:25]:
```suggestion
        os: [Ubuntu, MacOS, Windows]
```

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -8,120 +8,62 @@ jobs:
...
       - name: Linting
         run: |
           pip install pre-commit
           pre-commit run --all-files
 
-  Linux:
+  Tests:
     needs: Linting
-    runs-on: ubuntu-latest
+    runs-on: ${{ matrix.os }}
```
Comment (pradyunsg) [inline .github/workflows/main.yml:22]:
```suggestion
    name: ${{ matrix.os }} / ${{ matrix.python-version }}
    runs-on: ${{ matrix.os }}-latest
```

**Comment by abn** [comment]
Thanks @pradyunsg 

@sdispater added a pr for master as well, and updated the one for core too.

**Pull request merged**